ansible_base/lib/utils/timeout.py

```diff
@@ -0,0 +1,32 @@
+import signal
+
+
+class Timeout:
```

Comment:
In general I think that's an okay way to time out - careful about threads though, I don't know how this works in the context of uwsgi worker threads...

> Python signal handlers are always executed in the main Python thread of the main interpreter, even if the signal was received in another thread.

- https://docs.python.org/3/library/signal.html#signals-and-threads

I'm okay with us trying this, just something to watch out for ^

Reply:
I was able to track down why it didn't work before. We had a retry mechanism baked in. Removing that has helped although there are still some edge cases where we don't get the 4 sec timeout behavior. I'm not in love with that but its safer overall.